src/components/TrackList.js

```diff
@@ -58,6 +66,10 @@ TrackList.propTypes = {
   availableColors: PropTypes.array,
   onChange: PropTypes.func.isRequired,
   onDelete: PropTypes.func.isRequired,
+  showFileSizeAlert: PropTypes.func.isRequired,
+  setUploadInProgress: PropTypes.func.isRequired,
+  getPathNames: PropTypes.func.isRequired,
+  apiUrl: PropTypes.string.isRequired,
```

Comment:
And we're also punching all these props through several layers of components to get them to where they are used.